exercises/Stephen_Smith/ex02_oo_basics/include/Shapes.h

```diff
@@ -0,0 +1,218 @@
+#ifndef SHAPES_H_
+#define SHAPES_H_
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <math.h>  
+
+namespace{
+
+    const double PI=3.141592653589793238463;
+
+}
+
+
+class BaseShape{
+
+public:
+    
+    // Constructor
+    BaseShape();
+    virtual ~BaseShape() = 0 ;
+
+    // Getters
+    virtual int getNumSides() const = 0;
+    virtual std::string getShapeName() const =0;
+
+    // Return private members area and perimeter
+    double getArea() const;
+    double getPerimeter() const;
+
+    // Print shape information
+    virtual void printShapeDimensions() const = 0;
+
+    protected:
+        void setArea(double area) {this->m_area = area;}
+        void setPerimeter(double perimeter) {this->m_perimeter = perimeter;}
+
+    private:
+        double m_area;
+        double m_perimeter;
+
+};
+
+// Circle class
+class Circle : public BaseShape
```

Comment:
In is good practice to put each class in its own file. This avoid a client having to include more code than necessary if they just want a particular shape and it is also easier to find a particular class when looking for it as a developer.